src/transforms/lua/v2/mod.rs

```diff
@@ -52,12 +53,17 @@ pub struct LuaConfig {
     /// The program can be used to to import external dependencies, as well as define the functions
     /// used for the various lifecycle hooks. However, it's not strictly required, as the lifecycle
     /// hooks can be configured directly with inline Lua source for each respective hook.
+    #[configurable(metadata(
+        docs::examples = "function init()\n\tcount = 0\nend\n\nfunction process()\n\tcount = count + 1\nend\n\nfunction timer_handler(emit)\n\temit(make_counter(counter))\n\tcounter = 0\nend\n\nfunction shutdown(emit)\n\temit(make_counter(counter))\nend\n\nfunction make_counter(value)\n\treturn metric = {\n\t\tname = \"event_counter\",\n\t\tkind = \"incremental\",\n\t\ttimestamp = os.date(\"!*t\"),\n\t\tcounter = {\n\t\t\tvalue = value\n\t\t}\n \t}\nend",
```

Comment:
I agree that it's very ugly. This would/will be better if I add support to allow passing the value of a metadata attribute as an expression, so we could encapsulate this in a normal Rust function -- like what we do for defining default field values for `serde` -- and use multi-line strings, etc.